sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala

```diff
@@ -271,7 +271,7 @@ package object dsl {
       override def expr: Expression = Literal(s)
       def attr: UnresolvedAttribute = analysis.UnresolvedAttribute(s)
     }
-    implicit class DslAttr(attr: UnresolvedAttribute) extends ImplicitAttribute {
+    implicit class DslAttr(override val attr: UnresolvedAttribute) extends ImplicitAttribute {
```

Comment:
Shall we fix `ImplicitAttribute.attr`? It should be `analysis.UnresolvedAttribute(Seq(s))`